app/views/LocationController.js

```diff
@@ -13,102 +13,90 @@ import { SvgXml } from 'react-native-svg';
 import settingsIcon from './../assets/svgs/settingsIcon';
 import { isPlatformAndroid } from './../Util';
 import Colors from '../constants/colors';
-import LocationServices from '../services/LocationService';
-import DefaultPage from './tracking/DefaultPage';
-import ExposurePage from './tracking/ExposurePage';
-import OffPage from './tracking/OffPage';
-import styles from './tracking/style';
-import UnknownPage from './tracking/UnknownPage';
+import LocationServices, { Reason } from '../services/LocationService';
+import { DefaultPage } from './tracking/DefaultPage';
+import { ExposurePage } from './tracking/ExposurePage';
+import { OffPage } from './tracking/OffPage';
+import { styles } from './tracking/style';
+import { UnknownPage } from './tracking/UnknownPage';
 
-export default class LocationController extends Component {
-  constructor(props) {
-    super(props);
-
-    if (isPlatformAndroid()) {
-      StatusBar.setBackgroundColor(Colors.TRANSPARENT);
-      StatusBar.setBarStyle('light-content');
-      StatusBar.setTranslucent(true);
-    }
-
-    this.state = {
-      location: {
-        canTrack: true,
-        reason: '',
-        crossedPaths: false,
-      },
-    };
-  }
-
-  componentDidMount() {
-    this.updateStateInfo();
-    // refresh state if user backgrounds app
-    AppState.addEventListener('change', () => {
-      this.updateStateInfo();
-    });
-    // refresh state if settings change
-    this.unsubscribe = this.props.navigation.addListener('focus', () => {
-      this.updateStateInfo();
-    });
-    // handle back press
-    BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
+export const LocationController = props => {
+  if (isPlatformAndroid()) {
+    StatusBar.setBackgroundColor(Colors.TRANSPARENT);
+    StatusBar.setBarStyle('light-content');
+    StatusBar.setTranslucent(true);
   }
 
-  componentWillUnmount() {
-    console.log('unmounting component');
-    AppState.removeEventListener('change', this.handleAppStateChange);
-    BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
-    this.unsubscribe();
-  }
-
-  handleBackPress = () => {
-    BackHandler.exitApp(); // works best when the goBack is async
-    return true;
-  };
+  const [location, setLocation] = useState({
+    canTrack: true,
+    reason: '',
+    hasPotentialExposure: false,
+  });
 
-  getSettings() {
+  const SettingsIcon = () => {
```

Comment:
Move out into module scope... but I think this whole component be replaced with `<IconButton>` in the common components dir.